src/pages/writing-nft/index.vue

```diff
@@ -213,17 +216,49 @@ export default {
       this.$axios.$get(getTopNFTClasses()),
       this.$axios.$get(getLatestNFTClasses()),
     ]);
-    [this.trendingClassIds, this.latestClassIds] = [trendingRes, latestRes].map(
-      res =>
-        (res.classes || [])
-          .filter(
-            c => c.metadata?.nft_meta_collection_id === 'likerland_writing_nft'
-          )
-          .map(c => c.id)
-          .slice(0, 10)
+    const [trendingClasses, latestClasses] = [trendingRes, latestRes].map(res =>
+      (res.classes || []).filter(
+        c => c.metadata?.nft_meta_collection_id === 'likerland_writing_nft'
+      )
     );
+    this.latestClassIds = latestClasses
+      .slice(0, NFT_CLASS_DISPLAY_COUNT)
+      .map(c => c.id);
+    this.trendingClassIds = await this.filterNFTClassesByOwner(trendingClasses);
   },
   methods: {
+    async getOwnerFromUri(uri) {
+      try {
+        const data = await this.$axios.$get(uri);
+        return data.iscn_owner;
+      } catch (err) {
+        console.error(`Failed to fetch uri ${uri}`);
+        return null;
+      }
+    },
+    async filterNFTClassesByOwner(nftClasses) {
+      const ownerList = await Promise.all(
+        nftClasses
+          .slice(0, NFT_CLASS_DISPLAY_COUNT * 3)
```

Comment:
Why times 3? Should it be `MAX_NFT_CLASS_COUNT_PER_OWNER`?

Reply:
It's just an arbitrary estimated number. Since even we use `NFT_CLASS_DISPLAY_COUNT * MAX_NFT_CLASS_COUNT_PER_OWNER`, half of the works are probably from 2 or 3 owners.

Reply:
Maybe add a comment here

Reply:
Added